File: src/lambdas/calimaco/forgot-password.mjs
const CALIMACO_BASE_URL = process.env.CALIMACO_BASE_URL;

function isValidEmail(email) {
    const pattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
    return pattern.test(email);
}

function createResponse(statusCode, data) {
    return {
        statusCode,
        headers: {
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Headers": "*",
            "Access-Control-Allow-Methods": "OPTIONS,POST",
            "Content-Type": "application/json"
        },
        body: JSON.stringify(data)
    };
}

export const lambda_handler = async (event) => {
    try {
        const body = JSON.parse(event.body || "{}");
        const email = body.email || "";

        if (!email) {
            return createResponse(400, { message: "Completar los campos obligatorios" });
        }

        if (!isValidEmail(email)) {
            return createResponse(400, { message: "El email ingresado no tiene un formato válido" });
        }

        const response = await fetch(`${CALIMACO_BASE_URL}/auth/forgotPassword`, {
            method: "POST",
            headers: { "Content-Type": "application/x-www-form-urlencoded" },
            body: new URLSearchParams({
                company: "ACP",
                email
            }),
            signal: AbortSignal.timeout(10000) // 10 second timeout
        });

        const data = await response.json();
        return createResponse(response.status, data);

    } catch (error) {
        console.error("Error:", error);

        if (error.name === "TimeoutError" || error.name === "AbortError") {
            return createResponse(504, { message: "Timeout - No response from server" });
        }

        if (error instanceof SyntaxError) {
            return createResponse(400, { message: "Completar los campos obligatorios" });
        }

        return createResponse(500, { message: "Ocurrió un error inesperado" });
    }
};

File: src/lambdas/calimaco/forgot-password/forgot-password.py
import json
from json.decoder import JSONDecodeError
import requests
import re
import logging

logger = logging.getLogger()
logger.setLevel(logging.INFO)

def is_valid_email(email):
    pattern = r"(^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$)"
    return re.match(pattern, email) is not None


def lambda_handler(event, context):
    try:
        url = "https://api.casinoatlanticcity.com/api/auth/forgotPassword"

        # Obtener parámetros desde event
        body = json.loads(event.get("body") or "{}")

        company = "ACP"
        email = body.get("email", "")

        # Verificar si alguno es None
        if not email:
            return {
                "statusCode": 400,
                "headers": {
                    "Access-Control-Allow-Origin": "*",
                    "Access-Control-Allow-Headers": "*",
                    "Access-Control-Allow-Methods": "OPTIONS,POST",
                    "Content-Type": "application/json",
                },
                "body": json.dumps({"message": "Completar los campos obligatorios"}),
            }

        if not is_valid_email(email):
            return {
                "statusCode": 400,
                "headers": {
                    "Access-Control-Allow-Origin": "*",
                    "Access-Control-Allow-Headers": "*",
                    "Access-Control-Allow-Methods": "OPTIONS,POST",
                    "Content-Type": "application/json",
                },
                "body": json.dumps(
                    {"message": "El email ingresado no tiene un formato válido"}
                ),
            }

        payload = {"company": company, "email": email}

        headers = {"Content-Type": "application/x-www-form-urlencoded"}

        response = requests.post(url, data=payload, headers=headers, timeout=10)
        return {
            "statusCode": response.status_code,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "*",
                "Access-Control-Allow-Methods": "OPTIONS,POST",
                "Content-Type": "application/json",
            },
            "body": json.dumps(response.json()),
        }
    except requests.exceptions.Timeout:
        return {
            "statusCode": 504,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "*",
                "Access-Control-Allow-Methods": "OPTIONS,POST",
                "Content-Type": "application/json",
            },
            "body": json.dumps({"message": "Timeout - No response from server"}),
        }
    except AttributeError:
        return {
            "statusCode": 400,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "*",
                "Access-Control-Allow-Methods": "OPTIONS,POST",
                "Content-Type": "application/json",
            },
            "body": json.dumps({"message": "Completar los campos obligatorios"}),
        }
    except JSONDecodeError:
        return {
            "statusCode": 400,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "*",
                "Access-Control-Allow-Methods": "OPTIONS,POST",
                "Content-Type": "application/json",
            },
            "body": json.dumps({"message": "Completar los campos obligatorios"}),
        }
    except Exception as ex:
        logger.error("Unhandled exception: %s", str(ex))
        
        return {
            "statusCode": 500,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "*",
                "Access-Control-Allow-Methods": "OPTIONS,POST",
                "Content-Type": "application/json",
            },
            "body": json.dumps({"message": "Ocurrió un error inesperado"}),
        }
